clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractStickyAssignor.java

```diff
@@ -108,6 +107,8 @@ private boolean allSubscriptionsEqual(Set<String> allTopics,
                 subscribedTopics.addAll(subscription.topics());
             } else if (!(subscription.topics().size() == subscribedTopics.size()
                 && subscribedTopics.containsAll(subscription.topics()))) {
+                // we don't need consumerToOwnedPartitions in general assign case
+                consumerToOwnedPartitions = null;
```

Comment:
I'll separate the `checkAllSubscriptionEqual` and `getConsumerToOwnedPartitions` methods in KAFKA-12654

Reply:
I'll nullify it for now before invoking `generalAssign` to make the PR simple. And then adopt your 2nd option in another ticket (KAFKA-12654). Thank you.